### DISCUSSION TITLE
一些疑问

### DISCUSSION THREAD

**Discussion by yuantongziran** [XTLS/Xray-core#508] [Q&A] [answered]
如果前置的是web服务器，就可以在443端口实现比如www域名跳转root域名，不允许ip形式的https等等简单的功能。
但现在是vless前置，443端口不归web服务器管，这些web服务器的功能要怎么实现呢，好像没办法了吧。。。

**Comment by empter** [comment]
现在做法是 Nginx 监听在80端口，然后IP: your_server_ip访问统统跳转到https://your.domain.xyz
```
server {
    listen 127.0.0.1:80 default_server;
    ....
    ....
}

server {
    listen 127.0.0.1:80;
    server_name your_server_ip;
    return 301 https://your.domain.xyz$request_uri;
}

server {
    listen 0.0.0.0:80;
    server_name _;
    return 301 https://your.domain.xyz$request_uri;
}
```

不知道有没有更好的方式

**Reply by yuantongziran** [reply]
> 现在做法是 Nginx 监听在80端口，然后IP: your_server_ip访问统统跳转到https://your.domain.xyz
> 
> ```
> server {
>     listen 127.0.0.1:80 default_server;
>     ....
>     ....
> }
> 
> server {
>     listen 127.0.0.1:80;
>     server_name your_server_ip;
>     return 301 https://your.domain.xyz$request_uri;
> }
> 
> server {
>     listen 0.0.0.0:80;
>     server_name _;
>     return 301 https://your.domain.xyz$request_uri;
> }
> ```
> 
> 不知道有没有更好的方式

你没理解我的意思，80端口肯定是没问题的，但像是https://your_server_ip这种443端口的，我想做一个转发到https//example.com，或者直接返回403，这种要怎么办，sni也处理不了吧。

**Reply by galaxyskyknight** [reply]
当然可以处理了，xray fallback到Nginx, nginx用rewrite 和proxy_pass

**Comment by timi-owo** [comment, marked as answer]
1.前置 web server，用 sni 分流。
2.前置 xray，认证失败后的常规流量回落给 web server。

**Reply by yuantongziran** [reply]
> 1.前置 web server，用 sni 分流。
> 2.前置 xray，认证失败后的常规流量回落给 web server。

前置web server肯定是没问题的。
感谢提供思路，拜读了几篇相关的文章，已经解决。
